metricbeat/module/ceph/osd_df/data.go

```diff
@@ -0,0 +1,61 @@
+package osd_df
+
+import (
+	"encoding/json"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type Node struct {
+	ID          int64  `json:"id"`
+	Name        string `json:"name"`
+	Used        int64  `json:"kb_used"`
+	Available   int64  `json:"kb_avail"`
+	Total       int64  `json:"kb"`
+	PgNum       int64  `json:"pgs"`
+	DeviceClass string `json:"device_class"`
+}
+
+type Output struct {
+	Nodes []Node `json:"nodes"`
+}
+
+type OsdDfRequest struct {
+	Status string `json:"status"`
+	Output Output `json:"output"`
+}
+
+func eventsMapping(content []byte) ([]common.MapStr, error) {
+	var d OsdDfRequest
+	err := json.Unmarshal(content, &d)
+	if err != nil {
+		logp.Err("Error: ", err)
+		return nil, err
+	}
+
+	nodeList := d.Output.Nodes
+
+	//osd node list
+	events := []common.MapStr{}
+	for _, node := range nodeList {
+		nodeInfo := common.MapStr{}
```

Comment:
you could do here 
```
nodeInfo := common.MapStr{
  "id": node.ID,
  "name": node.Name,
...
```
I think it would make the code a bit nicer (exact same outcome)

Reply:
Good suggestion, your code looks much better. I will refine mine.